src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java

```diff
@@ -1,136 +1,183 @@
+    @Test
+    void givenCaseCannotBeListedRightAwayAndNoDirectionsBeingMade_thenNewHearingRequestSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = cannotBeListedAndNoDirectionsGiven();
+
+        assertHearingCreatedAndAdjournmentInProgress(response);
+    }
+
+    @DisplayName("When adjournment is enabled and case is LA and case can be listed right away "
+        + "then should send a new hearing request in hearings API")
+    @Test
+    void givenCanBeListedRightAway_thenNewHearingRequestSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = canBeListed();
+
+        assertHearingCreatedAndAdjournmentInProgress(response);
+    }
+
+    private void assertHearingCreatedAndAdjournmentInProgress(PreSubmitCallbackResponse<SscsCaseData> response) {
+        verify(hearingMessageHelper, times(1))
+            .sendListAssistCreateHearingMessage(sscsCaseData.getCcdCaseId());
+
+        assertThat(response.getErrors()).isEmpty();
+        assertThat(response.getData().getAdjournment().getAdjournmentInProgress()).isEqualTo(YES);
+    }
+
+    @DisplayName("When adjournment is enabled and case is LA and case cannot be listed right away "
+        + "and directions are being made, then should not send any messages")
+    @Test
+    void givenCaseCannotBeListedRightAwayAndDirectionsAreBeingMade_thenNoMessagesSent() {
+        PreSubmitCallbackResponse<SscsCaseData> response = cannotBeListedAndDirectionsGiven();
+
+        verifyNoInteractions(hearingMessageHelper);
+
+        assertThat(response.getErrors()).isEmpty();
+    }
+
+    @DisplayName("When we have changed the next hearing venue through an adjournment, show we change the region")
+    @Test
+    void givenAdjournCaseNextHearingVenueSelectedTrue_thenSetRegion() {
+        String venueId = "185";
+
+        RegionalProcessingCenter rpc = getRegionalProcessingCenter();
+        String postcode = rpc.getPostcode();
+
+        DynamicListItem venue = new DynamicListItem(venueId, null);
+        DynamicList adjournedNextVenue = new DynamicList(venue, null);
+        BenefitType benefitType = BenefitType.builder().code("PIP").build();
+
+        sscsCaseData.getAdjournment().setNextHearingVenueSelected(adjournedNextVenue);
+        sscsCaseData.setAppeal(Appeal.builder()
+            .appellant(Appellant.builder()
+                .address(Address.builder().postcode(postcode).build()).isAppointee(YES.getValue())
+                .build())
+            .benefitType(benefitType)
+            .build());
+
+        when(airLookupService.lookupAirVenueNameByPostCode(postcode, benefitType)).thenReturn("cardiff");
+        when(regionalProcessingCenterService.getByVenueId(venueId)).thenReturn(rpc);
+
+        PreSubmitCallbackResponse<SscsCaseData> response =
+            handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getErrors()).isEmpty();
+
+        assertThat(sscsCaseData.getRegion()).isEqualTo(rpc.getCity());
```

Comment:
I think this is better, though I am still having to struggle to make sense of it. I believe it would still benefit from earlier on proving that the region is != rpc.getCity. 
The general structure of a test should be a clear GIVEN, WHEN, THEN progression. In this case, I would like to see:
GIVEN - stuff & region == X
WHEN - change venue
THEN - region == Y

As it is right now, as a reviewer I am having to work out WHY this result is correct, whereas you could simply show me with an earlier assert.

Reply:
case data region at the start is set to SUTTON so you can clearly see it change. As well as that, it is now clear the difference between the processing center and region